src/cares_wrap.cc

```diff
@@ -121,10 +121,12 @@ inline const char* ToErrorCodeString(int status) {
 
 class ChannelWrap;
 
-struct node_ares_task {
+struct node_ares_task : public MemoryRetainer {
   ChannelWrap* channel;
   ares_socket_t sock;
   uv_poll_t poll_watcher;
+
+  void MemoryInfo(MemoryTracker* tracker) const override;
```

Comment:
Shouldn't these subclasses of `MemoryRetainer` override their `MemoryInfoName()` as well?

Reply:
> If I am reading the code correctly, this prefers using the class names of the target nodes as edge names?

Yes… I don’t think it makes much difference in this case, at least as long as we can’t name edges?